cmd/blockchaincmd/add_validator.go

```diff
@@ -252,15 +257,43 @@ func addValidator(cmd *cobra.Command, args []string) error {
 	}
 
 	if sovereign {
+		totalWeight, err := validator.GetTotalWeight(network.SDKNetwork(), subnetID)
+		if err != nil {
+			return err
+		}
+		validatorManagerRPCEndpoint := sc.Networks[network.Name()].ValidatorManagerRPCEndpoint
+		validatorManagerAddress := sc.Networks[network.Name()].ValidatorManagerAddress
+		if validatorManagerRPCEndpoint == "" {
+			return fmt.Errorf("unable to find Validator Manager RPC endpoint")
+		}
+		if validatorManagerAddress == "" {
+			return fmt.Errorf("unable to find Validator Manager address")
+		}
+		churnSettings, err := validatormanagersdk.GetChurnSettings(
+			validatorManagerRPCEndpoint,
+			common.HexToAddress(validatorManagerAddress),
+		)
+		if err != nil {
+			return err
+		}
+		allowedChange := float64(totalWeight*uint64(churnSettings.MaximumChurnPercentage)) / 100.0
 		if !cmd.Flags().Changed(validatorWeightFlag) {
 			weight, err = app.Prompt.CaptureWeight(
-				"What weight would you like to assign to the validator?",
-				func(uint64) error { return nil },
+				fmt.Sprintf("What weight would you like to assign to the validator (max=%d)?", uint64(allowedChange)),
+				func(weight uint64) error {
+					if float64(weight) > allowedChange {
+						return fmt.Errorf("can't make change: desired validator weight %d exceeds max allowed weight change of %d", weight, uint64(allowedChange))
```

Comment:
can we pls have a test in e2e that tests this? where we try to add a validator with weight more than allowed